crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -638,22 +667,26 @@ impl MeshPipeline {
 }
 
 impl GetBatchData for MeshPipeline {
-    type Query = (
-        Option<&'static MaterialBindGroupId>,
-        &'static Handle<Mesh>,
-        &'static MeshTransforms,
-    );
-    type CompareData = (Option<MaterialBindGroupId>, AssetId<Mesh>);
+    type Param = SRes<RenderMeshInstances>;
+    type Query = Entity;
+    type QueryFilter = With<Mesh3d>;
+    type CompareData = (MaterialBindGroupId, AssetId<Mesh>);
     type BufferData = MeshUniform;
 
-    fn get_buffer_data(&(.., mesh_transforms): &QueryItem<Self::Query>) -> Self::BufferData {
-        mesh_transforms.into()
-    }
-
-    fn get_compare_data(
-        &(material_bind_group_id, mesh_handle, ..): &QueryItem<Self::Query>,
-    ) -> Self::CompareData {
-        (material_bind_group_id.copied(), mesh_handle.id())
+    fn get_batch_data(
+        mesh_instances: &SystemParamItem<Self::Param>,
+        entity: &QueryItem<Self::Query>,
+    ) -> (Self::BufferData, Option<Self::CompareData>) {
+        let mesh_instance = mesh_instances
+            .get(entity)
+            .expect("Failed to find render mesh instance");
+        (
+            (&mesh_instance.transforms).into(),
+            mesh_instance.automatic_batching.then_some((
```

Comment:
Maybe instead of using `automatic_batching`, check if it has some components denoting it is skinned and/or morphed?

Reply:
I don't want to do more per-instance lookups of skin/morph data because that will hurt batching performance.